evproc/processor.py

```diff
@@ -308,11 +311,16 @@ def process(self, ev):
                   ``StopProcessing`` exception raised by a processor.
         """
 
-        # Walk through all the processors and process the event
-        for proc in self._get_procs(ev):
-            try:
-                proc(ev)
-            except exc.StopProcessing as ex:
-                return ex.retval
+        event_deque = collections.deque([ev])
+        while event_deque:
+            ev = event_deque.popleft()
+            # Walk through all the processors and process the event
+            for proc in self._get_procs(ev):
+                try:
+                    additional_events = proc(ev)
+                except exc.StopProcessing as ex:
+                    return ex.retval
```

Comment:
Let's not change this into a generator, as that requires a backwards-incompatible change; that means we have to do a return.  I say we should only do a return if we're stopping all event processing.

In fact, I'm tempted to suggest that we should only return something if the initial event raises a StopProcessing with the appropriate settings, and ignore return code in any other StopProcessing exception.  The mechanism is intended to not just allow event processing to be stopped, but to also potentially allow a value for the event processing to be returned to whatever launched the event in the first place; in that case, an alternative event might be an entirely different interface.